server/server.js

```diff
@@ -1,15 +1,370 @@
+    console.log("Collection deleted");
+
+    //insert the videos into the collection
+    const initialVideos = [
+      {
+        id: 523523,
+        title: "Never Gonna Give You Up",
+        url: "https://www.youtube.com/embed/dQw4w9WgXcQ",
+        rating: 23,
+      },
+      {
+        id: 523427,
+        title: "The Coding Train",
+        url: "https://www.youtube.com/embed/HerCR8bw_GE",
+        rating: 230,
+      },
+      {
+        id: 82653,
+        title: "Mac & Cheese | Basics with Babish",
+        url: "https://www.youtube.com/embed/FUeyrEN14Rk",
+        rating: 2111,
+      },
+      {
+        id: 858566,
+        title: "Videos for Cats to Watch - 8 Hour Bird Bonanza",
+        url: "https://www.youtube.com/embed/xbs7FT7dXYc",
+        rating: 11,
+      },
+      {
+        id: 453538,
+        title:
+          "The Complete London 2012 Opening Ceremony | London 2012 Olympic Games",
+        url: "https://www.youtube.com/embed/4As0e4de-rI",
+        rating: 3211,
+      },
+      {
+        id: 283634,
+        title: "Learn Unity - Beginner's Game Development Course",
+        url: "https://www.youtube.com/embed/gB1F9G0JXOo",
+        rating: 211,
+      },
+      {
+        id: 562824,
+        title: "Cracking Enigma in 2021 - Computerphile",
+        url: "https://www.youtube.com/embed/RzWB5jL5RX0",
+        rating: 111,
+      },
+      {
+        id: 442452,
+        title: "Coding Adventure: Chess AI",
+        url: "https://www.youtube.com/embed/U4ogK0MIzqk",
+        rating: 671,
+      },
+      {
+        id: 536363,
+        title: "Coding Adventure: Ant and Slime Simulations",
+        url: "https://www.youtube.com/embed/X-iSQQgOd1A",
+        rating: 76,
+      },
+      {
+        id: 323445,
+        title: "Why the Tour de France is so brutal",
+        url: "https://www.youtube.com/embed/ZacOS8NBK6U",
+        rating: 73,
+      },
+    ];
+    await collection.insertMany(initialVideos);
+    console.log("Initial videos inserted");
+
+    app.get("/", async (req, res) => {
+      const videos = await collection.find().toArray();
+      res.send(videos);
+    });
+
+    app.post("/", async (req, res) => {
+      const { title, url } = req.body;
+
+      if (!title || !url) {
+        return res.status(400).json({
+          result: "failure",
+          message: "Video could not be saved",
+        });
+      } else {
+        const maxIdVideo = await collection.findOne({}, { sort: { id: -1 } });
+
+        let newId = 1;
+        if (maxIdVideo) {
+          newId = maxIdVideo.id + 1;
+        }
+        const newVideo = {
+          id: newId,
+          title,
+          url,
+          rating: 0,
+        };
 
-// Store and retrieve your videos from here
-// If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+        await collection.insertOne(newVideo);
+        res.status(201).send(`The new video has been added`);
```

Comment:
Anything else that could respond with, just 400 and 200?

What would happen if mongo crashed?